msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/GetAccessTokenTests.kt

```diff
@@ -36,11 +37,32 @@ import org.robolectric.annotation.Config
 
 @Config(shadows = [ShadowBaseController::class])
 class GetAccessTokenTests : NativeAuthPublicClientApplicationAbstractTest() {
+    private val INVALID_SCOPE = NativeAuthCredentialHelper.nativeAuthInvalidScope
     private val EMPLOYEE_WRITE_ALL_SCOPE = NativeAuthCredentialHelper.nativeAuthEmployeeWriteAllScope
     private val EMPLOYEE_READ_ALL_SCOPE = NativeAuthCredentialHelper.nativeAuthEmployeeReadAllScope
     private val CUSTOMERS_WRITE_ALL_SCOPE = NativeAuthCredentialHelper.nativeAuthCustomerWriteAllScope
     private val CUSTOMERS_READ_ALL_SCOPE = NativeAuthCredentialHelper.nativeAuthCustomerReadAllScope
 
+    /**
+     * Signing in with an invalid scope should make the API and the SDK return an error.
+     */
+    @Test
+    fun testGetAccessTokenForInvalidScope() = runTest {
+        val username = NativeAuthCredentialHelper.nativeAuthSignInUsername
+        val password = getSafePassword()
+        val result = application.signIn(
+            username = username,
+            password = password.toCharArray(),
+            scopes = listOf(INVALID_SCOPE)
+        )
+        assertState<SignInError>(result)
+        Assert.assertEquals("invalid_grant", (result as SignInError).error)
+        Assert.assertNotNull(result.errorMessage)
+        Assert.assertTrue(result.errorMessage!!.contains("AADSTS65001: The user or administrator has not consented to use the application"))
```

Comment:
Does "AADSTS65001" a fixed string from server? Probably error message checking is not a necessity.